### PR TITLE
mlflow is not required if not being used

### PR DESCRIPTION
**Pull request by williamFalcon**
remove requirement to import mlflow

### CODE REVIEW

**Comment by williamFalcon** [comment]
@neggert users shouldn't have to install mlflow or comet if they're not using it.

**Comment by neggert** [comment]
Yeah, was just about to send a PR when I saw this. It was originally this way, but got messed up in #395. I have a different fix that I like better because it still throws an error if you explicitly try to import e.g. CometLogger without Comet installed.

File: pytorch_lightning/logging/mlflow_logger.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 try:
     import mlflow
-except ImportError:
-    raise ImportError('Missing mlflow package.')
+except Exception as e:
```
Comment (Borda) [inline pytorch_lightning/logging/mlflow_logger.py]: you do not need the `e`, PEP8 will complain that you define a variable which is not used

**Comment by Borda** [comment]
@williamFalcon @neggert I do not agree with ignoring the import error. when you try to import `mlflow_logger.py` directly and you do not have `mlflow` you should get an error message... The ignore is done in `logging.__init__.py` where you say the `mlflow` or any other is not needed.

**Comment by neggert** [comment]
@Borda I think #442 does what you're asking for.

**Comment by Borda** [comment]
> @Borda I think #442 does what you're asking for.

That one is fine, but with this PR you change the meaning of `ImportError` completely...
It is like you go to sweet shop with optional desk with ice-screen, you want ice -cream, go to the desk and it is empty without any information... keeping the raise Exception gives you info that there is no ice-cream if you come to the deck but it does not border you if you come to sweet shop and you do not case about ice-cream...

**Pull request closed without merging**